featuretools/feature_base/features_serializer.py

```diff
@@ -96,11 +100,36 @@ def save(self, location, profile_name):
     def _feature_definitions(self):
         if not self._features_dict:
             self._features_dict = {}
+            self._primitives_dict = {}
 
             for feature in self.feature_list:
                 self._serialize_feature(feature)
 
-        return self._features_dict
+            primitive_number = 0
+            primitive_id_to_key = {}
+            for name, feature in self._features_dict.items():
+                primitive = feature["arguments"].get("primitive")
+                if primitive:
+                    primitive_id = id(primitive)
+                    if primitive_id not in primitive_id_to_key.keys():
+                        # Primitive we haven't seen before, add to dict and increment primitive_id counter
+                        # Always use string for keys because json conversion results in integer dict keys
+                        # being converted to strings, but integer dict values are not.
+                        primitives_dict_key = str(primitive_number)
+                        primitive_id_to_key[primitive_id] = primitives_dict_key
+                        self._primitives_dict[
+                            str(primitives_dict_key)
```

Comment:
`primitives_dict_key` is already cast to string on line 118

Reply:
Yep, sure is. Fixed [9e6830c](https://github.com/alteryx/featuretools/pull/2144/commits/9e6830c89e9d02d851cc11483039cdf20627a2a2)